client/blocks/site-preview/index.jsx

```diff
@@ -1,110 +1,62 @@
 
-const debug = debugFactory( 'calypso:site-preview' );
+function SitePreviewInner( { siteId, className } ) {
+	const dispatch = useDispatch();
+	const [ previewCount, incrementPreviewCount ] = useReducer( ( n ) => n + 1, 0 );
 
-class SitePreview extends Component {
-	static propTypes = {
-		className: PropTypes.string,
-		showPreview: PropTypes.bool,
-		previewUrl: PropTypes.string,
-		selectedSite: PropTypes.object,
-		selectedSiteId: PropTypes.number,
-		selectedSiteNonce: PropTypes.string,
-		selectedSiteUrl: PropTypes.string,
-		closePreview: PropTypes.func.isRequired,
-	};
+	const showPreview = useSelector( ( state ) => getCurrentLayoutFocus( state ) === 'preview' );
+	const selectedSiteNonce = useSelector( ( state ) =>
+		getSiteOption( state, siteId, 'frame_nonce' )
+	);
+	const previewUrl = useSelector( getPreviewUrl );
+	const hideSEO = useSelector( ( state ) => isDomainOnlySite( state, siteId ) );
 
-	state = {
-		previewCount: 0,
-	};
-
-	previewCounter = 0;
-
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( this.props.selectedSiteId && this.props.selectedSiteId !== nextProps.selectedSiteId ) {
-			this.previewCounter = 0;
-		}
-
-		if ( ! this.props.showPreview && nextProps.showPreview ) {
-			debug( 'forcing refresh' );
-			this.previewCounter > 0 && this.setState( { previewCount: this.previewCounter } );
-			this.previewCounter += 1;
-		}
-	}
-
-	getPreviewUrl() {
-		if ( ! this.props.selectedSiteUrl && ! this.props.previewUrl ) {
-			debug( 'no preview url and no site url were found for this site' );
+	function formatPreviewUrl() {
+		if ( ! previewUrl ) {
 			return null;
 		}
-		const previewUrl = addQueryArgs(
+
+		return addQueryArgs(
 			{
 				iframe: true,
 				theme_preview: true,
-				'frame-nonce': this.props.selectedSiteNonce,
-				cachebust: this.state.previewCount,
+				'frame-nonce': selectedSiteNonce ?? '',
+				cachebust: previewCount,
 			},
-			this.getBasePreviewUrl()
+			previewUrl
 		);
-		debug( 'using this preview url', previewUrl );
-		return previewUrl;
 	}
 
-	getBasePreviewUrl() {
-		return this.props.previewUrl || this.props.selectedSiteUrl;
-	}
-
-	render() {
-		if ( ! this.props.selectedSitePreviewable ) {
-			debug( 'a preview is not available for this site' );
-			return null;
-		}
-
-		return (
-			<WebPreview
-				className={ this.props.className }
-				previewUrl={ this.getPreviewUrl() }
-				externalUrl={ this.getBasePreviewUrl() }
-				showExternal={ true }
-				showClose={ true }
-				showPreview={ this.props.showPreview }
-				onClose={ this.props.closePreview }
-				showSEO={ ! this.props.isDomainOnlySite }
-			/>
-		);
-	}
+	return (
+		<WebPreview
+			className={ className }
+			previewUrl={ formatPreviewUrl() }
+			externalUrl={ previewUrl }
+			showExternal={ true }
+			showClose={ true }
```

Comment:
Now that we're here, we can also clean those arbitrary bools up:

```suggestion
			showExternal
			showClose
```